crates/marketplace/src/builder_state.rs

```diff
@@ -424,8 +437,45 @@ impl<TYPES: NodeType> BuilderState<TYPES> {
         // Case 1: Bootstrapping phase
         // Case 2: No intended builder state exist
         // To handle both cases, we can have the highest view number builder state running
-        // and only doing the insertion if and only if intended builder state for a particulat view is not present
+        // and only doing the insertion if and only if intended builder state for a particular view is not present
         // check the presence of quorum_proposal.data.view_number-1 in the spawned_builder_states list
+        if qc_msg.proposal.data.justify_qc.view_number != self.built_from_proposed_block.view_number
```

Comment:
Hmm... You added this back in?  I had removed this primarily due to handling similar logic with the `am_i_the_best_builder_state_to_extend` logic, and to make the `process_quorum_proposal` and `process_da_proposal` have similar logic paths.

I think with this present, some of the order tests don't pass.

Reply:
oops sorry I think this is because I didn't look carefully when doing conflict resolving, let me recheck all these